Dockerfile

```diff
@@ -41,36 +41,34 @@ RUN apt-get clean \
   && apt-get update \
   && apt-get install -y \
   build-essential python3.9 python3-pip git cmake make ruby rake autoconf automake \
-  && pip install conan -v "conan==1.59.0"
+  && pip install conan -v "conan==2.0.9"
 
 # make an agent directory and cd into it
 WORKDIR /root/agent
 
 # bring in the repo contents, minus .dockerignore files
 COPY . .
 
-# set some variables
-ENV PATH=$HOME/venv3.9/bin:$PATH
-ENV CONAN_PROFILE=conan/profiles/docker
-ENV WITH_RUBY=True
-ENV WITH_TESTS=False
+ARG WITH_RUBY=True
+ARG WITH_TESTS=False
 
 # limit cpus so don't run out of memory on local machine
 # symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
 # can turn off if building in cloud
-ENV CONAN_CPU_COUNT=1
+ARG CONAN_CPU_COUNT=2
 
-# make installer
-RUN conan export conan/mqtt_cpp \
-  && conan export conan/mruby \
-  && conan install . -if build --build=missing \
-  -pr $CONAN_PROFILE \
-  -o build_tests=$WITH_TESTS \
-  -o run_tests=$WITH_TESTS \
-  -o with_ruby=$WITH_RUBY
 
-# compile source (~20mins - 4hrs for qemu)
- RUN conan build . -bf build
+# set some variables
+ENV PATH=$HOME/venv3.9/bin:$PATH
+ENV CONAN_PROFILE=conan/profiles/docker
+
+# make installer
+RUN conan profile detect
+RUN conan install . --build=missing -pr $CONAN_PROFILE -o with_ruby=$WITH_RUBY -c tools.build:skip_test=True \
+     -c tools.build:jobs=$CONAN_CPU_COUNT
+RUN conan create . --build=missing -pr $CONAN_PROFILE -o with_ruby=$WITH_RUBY -o cpack=True \
```

Comment:
@wsobel, do we really need the `with_ruby` variable to be defined in `Dockerfile`? Wouldn’t it be better to move it to `conan/profiles/docker`?

Why do you skip tests with `-c tools.build:skip_test=True`?

Reply:
This allows you to change it with the --build-arg instead of creating a new profile. We could have user supplied profiles, but I'm not sure how effective that would be. We can consider.

Reply:
> This allows you to change it with the --build-arg instead of creating a new profile.

Is there a `conan` env var to override `skip_test` value / force run tests? I couldn’t find one in the [docs](https://docs.conan.io/en/1.60/reference/env_vars.html).

> We could have user supplied profiles, but I'm not sure how effective that would be.

Well, I think that _most_ will use the image published on Docker Hub anyway. However, we could use something like `${CONAN_PROFILE:-conan/profiles/docker}` without using `ENV` in `Dockerfile`; then it could be overridden using `--build-arg CONAN_PROFILE='path/to/profile'`.

Reply:
You can set the global conf setting or set it in the default profile. That would be the closest to what you are referring to. The issue is without the `-tf` switch, it still goes through the build of the test_package. I have added the `-o development=True` to make the tests more like the previous build behavior. 

I added development so the tests were build as a subfolder in cmake and integrated into the IDEs with a simpler workflow. I've been working on making the builds multi-modal.

We may also need different profiles for other OS flavors.